### ISSUE TITLE
Blockwise transfer fails when block number exceeds 15

### ISSUE THREAD

**Issue by JKRhb** [shamblett/coap#47]
When performing a (Block2) blockwise transfer, the block numbering seems to jump to 128 instead of 16 after the block with number 15 has been received. At the same time, (in my case) the block size jumps from 16 to 32 indicating that there seems to be an issue with the binary encoding. I will try to provide a code example for reproducing the issue.

**Comment by shamblett** [comment]
Thanks. There is an example for this in the examples directory,  blockwise_xfer.dart, this works but only receives 13 blocks so it doesn't show the error.

**Comment by JKRhb** [comment]
Thank you for your quick response! I think I now managed to find the cause of the issue: There is actually an `_compressValueBytes` method in the `CoapBlockOption` class (exposed through a `blockValueBytes` getter) which implements the correct encoding behavior. During serialization, however, only the regular `valueBytes` getter is called which is the reason why the encoded option gets malformed. In the tests the `blockValueBytes` is being used which is why the unintended behavior hasn't been detected so far.

I think I have already been able to come up with a fix for which I will open a PR soon :)
